corehq/messaging/smsbackends/infobip/views.py

```diff
@@ -0,0 +1,32 @@
+import json
+from corehq.apps.sms.api import incoming as incoming_sms
+from corehq.apps.sms.views import IncomingBackendView
+from corehq.messaging.smsbackends.infobip.models import SQLInfobipBackend
+from django.http import HttpResponse
+
+
+class InfobipIncomingMessageView(IncomingBackendView):
+    urlname = 'infobip_message'
+
+    @property
+    def backend_class(self):
+        return SQLInfobipBackend
+
+    def post(self, request, api_key, *args, **kwargs):
+        request_body = json.loads(request.body)
+        for message in request_body.get('results'):
+            message_sid = message.get('messageId')
+            from_ = message.get('from')
+            message_content = message.get('message')
+            if message_content.get('type') == 'TEXT':
+                body = message_content.get('text', {})
+
+        incoming_sms(
```

Comment:
I think this should be indented into the `for` loop? Otherwise you'll only process the last message.

Reply:
Although it is very similar to turn, the request payload has slightly different format for infobip. I am already looping through all received messages here:
 **for message in request_body.get('results'):**

here is the payload format for infobip:
{
  "results": [
    {
      "from": "385919998888",
      "to": "41793026731",
      "integrationType": "WHATSAPP",
      "receivedAt": "2019-07-19T11:23:26.998+0000",
      "messageId": "ABEGOFl3VCQoAhBalbc6rTQT6mgS29EmGZ7a",
      "message": {
        "type": "TEXT",
        "text": "Support hello"
      },
      "contact": {
        "name": "Frank"
      },
      "price": {
        "pricePerMessage": 0,
        "currency": "EUR"
      }
    }
  ],
  "messageCount": 1,
  "pendingMessageCount": 0
}

Reply:
Maybe I'm just reading this wrong, but I think if you have a `messageCount` higher than 1 (i.e. more than 1 message in your results list), then you will only call `incoming_sms` on the last one as it executes after the for loop has completed. I think you want to run `incoming_sms` on every message, right?

Reply:
You are completely right. Now I see. I have to indent the **incoming_sms** method call. I've never received multiple messages in the payload while testing and hence I didn't realize that I made a mistake.

Reply:
It will be fixed with my next update. Thanks